File: src/game/engine/GameTickerObserver.java
package game.engine;

import game.events.Observer2;

import java.util.Observable;
import java.util.Observer;

public abstract class GameTickerObserver implements Observer2 {
    private static final float DEFAULT_DELAY_OBSERVER = 40;
    protected long lastUpdate = 0L;

    @Override
    public void update(Object arg) {
        lastUpdate = System.currentTimeMillis();
    }

    public long getLastUpdate() {
        return lastUpdate;
    }

    protected float getDelayObserverUpdate(){
        return DEFAULT_DELAY_OBSERVER;
    }
}

File: src/game/engine/tasks/GameTickerObserver.kt
package game.engine.tasks

import game.engine.tasks.observer.Observer2
import game.utils.time.now

abstract class GameTickerObserver : Observer2 {
    var lastUpdate = 0L
        protected set

    override fun update(arg: Any?) {
        lastUpdate = now()
    }

    open val delayObserverUpdate: Float
        get() {
            return DEFAULT_OBSERVER_UPDATE
        }

    companion object {
        val DEFAULT_OBSERVER_UPDATE = 30f
    }
}